js/main.js

```diff
@@ -56,7 +57,14 @@ function deckLoad(event) {
 function cardInDeckDetails(event) {
   if (event.target.tagName === 'IMG') {
     getYugiohDataExact(event.target.name);
-    $singleView.className = 'container single-view';
+    setTimeout(function () {
+      $loadingAnimation.className = 'row loading';
+      $singleView.className = 'hidden';
+    }, 0);
+    setTimeout(function () {
+      $loadingAnimation.className = 'row loading hidden';
+      $singleView.className = 'container single-view';
+    }, 1000);
```

Comment:
You shouldn't handle loading spinners using a timeout, that means you run the risk of waiting far too long, or not long enough. Instead, you should show the spinner when you send a request, and hide it again when the `load` event handler runs.